mapper/orm/src/main/java/org/hibernate/search/mapper/orm/bootstrap/impl/HibernateSearchCompositeMappingProducer.java

```diff
@@ -6,51 +6,52 @@
-import org.hibernate.search.util.common.annotation.impl.SuppressForbiddenApis;
 
-import org.jboss.jandex.IndexView;
+public class HibernateSearchCompositeMappingProducer implements AdditionalMappingContributor {
 
-@SuppressWarnings("deprecation")
-public class HibernateSearchCompositeMappingProducer implements org.hibernate.boot.spi.AdditionalJaxbMappingProducer {
+	@Override
+	public String getContributorName() {
+		return "hibernate-search";
+	}
 
 	@Override
-	@SuppressForbiddenApis(reason = "Strangely, this SPI involves the internal MappingBinder class,"
-			+ " and there's nothing we can do about it")
-	public Collection<MappingDocument> produceAdditionalMappings(final MetadataImplementor metadata,
-			IndexView jandexIndex, final MappingBinder mappingBinder, final MetadataBuildingContext buildingContext) {
+	public void contribute(AdditionalMappingContributions contributions, InFlightMetadataCollector metadata,
+			ResourceStreamLocator resourceStreamLocator, MetadataBuildingContext buildingContext) {
 		Optional<HibernateSearchPreIntegrationService> preIntegrationServiceOptional =
 				HibernateOrmUtils.getServiceOrEmpty( buildingContext.getBootstrapContext().getServiceRegistry(),
 						HibernateSearchPreIntegrationService.class );
-		if ( !preIntegrationServiceOptional.isPresent() ) {
-			// Hibernate Search is disabled
-			return Collections.emptyList();
-		}
+		if ( preIntegrationServiceOptional.isPresent() ) {
+			HibernateSearchPreIntegrationService preIntegrationService = preIntegrationServiceOptional.get();
+
+			ConfigurationPropertySource propertySource = preIntegrationService.propertySource()
+					.withMask( SearchIntegrationEnvironment.CONFIGURATION_PROPERTIES_MASK )
+					.withMask( HibernateOrmMapperSettings.Radicals.COORDINATION );
 
-		HibernateSearchPreIntegrationService preIntegrationService = preIntegrationServiceOptional.get();
-		List<MappingDocument> mappings = new ArrayList<>();
-		ConfigurationPropertySource propertySource = preIntegrationService.propertySource()
-				.withMask( SearchIntegrationEnvironment.CONFIGURATION_PROPERTIES_MASK )
-				.withMask( HibernateOrmMapperSettings.Radicals.COORDINATION );
-		for ( HibernateSearchOrmMappingProducer mappingProducer : preIntegrationService
-				.coordinationStrategyConfiguration().mappingProducers() ) {
-			mappings.addAll( mappingProducer.produceMappings( propertySource, metadata.getDatabase().getDialect(),
-					mappingBinder, buildingContext ) );
+			for ( HibernateSearchOrmMappingProducer mappingProducer : preIntegrationService
+					.coordinationStrategyConfiguration().mappingProducers() ) {
+				for ( Map.Entry<Class<?>, JaxbEntityMappings> entry : mappingProducer.produceMappings(
+						propertySource,
+						metadata.getDatabase().getDialect(),
+						buildingContext
+				).entrySet() ) {
+					contributions.contributeEntity( entry.getKey() );
+					contributions.contributeBinding( entry.getValue() );
+				}
+			}
```

Comment:
And this diff right here is why one doesn't rework indentation of a whole method just because one doesn't like early returns :D